edge-core/protocol_api.c

```diff
@@ -1140,98 +1080,60 @@ int write_to_pt_fota(edgeclient_request_context_t *request_ctx, void *userdata)
              request_ctx->object_instance_id,
              request_ctx->resource_id);
 
-    arm_uc_buffer_t manifest_buffer, url_buffer, hash_buffer;
-    manifest_buffer.ptr = request_ctx->value;
-    manifest_buffer.size = request_ctx->value_len;
-    manifest_buffer.size_max = SN_COAP_MAX_BLOCKWISE_PAYLOAD_SIZE;
-
-    uint32_t fw_size = 0;
-    uint64_t fw_version = 0;
-
-    //Check Vendor ID
-    arm_uc_buffer_t manifest_vendor_id, manifest_class_id;
-    arm_uc_error_t error = ARM_UC_mmGetVendorGuid(&manifest_buffer,&manifest_vendor_id);
-    if (error.code != ERR_NONE) {
-        tr_error("ERROR getting vendor ID from manifest: %d", error.code);
-        return 1;
-    }
-
-    //Class ID Check
-    error = ARM_UC_mmGetClassGuid(&manifest_buffer,&manifest_class_id);
-    if (error.code != ERR_NONE) {
-        tr_error("ERROR getting class ID from manifest: %d", error.code);
-        return 1;
-    }
-
-    //firmware size check
-    error = ARM_UC_mmGetFwSize(&manifest_buffer, &(fw_size));
-    if (error.code != ERR_NONE) {
-        tr_error("ERROR getting firmware size: %d", error.code);
-        return 1;
-    }
-
-    // firmware url
-    error = ARM_UC_mmGetFwUri(&manifest_buffer,&url_buffer );
-    if (error.code != ERR_NONE) {
-        tr_error("ERROR getting firmware URI: %d", error.code);
-        return 1;
-    }
-
-    error = ARM_UC_mmGetFwHash(&manifest_buffer, &(hash_buffer));
-    if (error.code != ERR_NONE) {
-        tr_error("ERROR getting firmware hash: %d", error.code);
-        return 1;
-    }
-
-    error = ARM_UC_mmGetTimestamp(&manifest_buffer, &fw_version);
-    if (error.code != ERR_NONE) {
-        tr_error("ERROR getting firmware version: %d", error.code);
-        return 1;
-    }
+    uint8_t manifest_vendor_id[VENDOR_ID_SIZE] = {0};
+    uint8_t manifest_class_id[CLASS_ID_SIZE] = {0};
+    char uri[MANIFEST_URI_SIZE] = {0};
+    char new_version[FOTA_COMPONENT_MAX_STR_SIZE] = "";
+    char old_version[FOTA_COMPONENT_MAX_STR_SIZE] = "";
+    char component[MAX_FOTA_STR] = "";
+    get_class_id(manifest_class_id);
+    get_vendor_id(manifest_vendor_id);
+    get_uri(uri);
+    get_component_name(component);
```

Comment:
@prakharbhatt007 lets verify that the pointer is not empty.

Reply:
Implemented